provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/practitioner/PractitionerController.java

```diff
@@ -148,13 +138,111 @@ private PractitionerWrapper search(MultiValueMap<String, String> parameters) {
+        providerResponsePages
+            .parallelStream()
+            .map(
+                prv -> {
+                  return ppmsClient.providerContactsForId(prv.providerIdentifier().toString());
+                })
+            .collect(Collectors.toList());
+    /**
+     * Wrap providerResponse and providerContacts together to create a list of Practitioner (FHIR).
+     */
+    List<PractitionerWrapper> practitionerWrapperPages =
+        IntStream.range(0, providerContactsResponsePages.size())
+            .parallel()
+            .mapToObj(
+                i ->
+                    practitionerWrapperBuilder(
+                        providerContactsResponsePages.get(i), providerResponsePages.get(i)))
+            .collect(Collectors.toList());
+    return Pair.of(practitionerWrapperPages, providerResponsePages.size());
+  }
+
+  /** Logic for search by Identifier. */
+  private Pair<List<PractitionerWrapper>, Integer> searchIdentifier(
+      MultiValueMap<String, String> parameters) {
+    PractitionerWrapper.PractitionerWrapperBuilder practitionerWrapper =
+        new PractitionerWrapper.PractitionerWrapperBuilder();
+    String identifier = parameters.getFirst("identifier");
+    ProviderResponse providerResponse = ppmsClient.providersForId(identifier);
+    String providerIdentifier = providerResponse.value().get(0).providerIdentifier().toString();
+    ProviderContactsResponse providerContactsResponse =
+        ppmsClient.providerContactsForId(providerIdentifier);
+    return Pair.of(
+        singletonList(
+            practitionerWrapper
+                .providerContactsResponse(providerContactsResponse)
+                .providerResponse(providerResponse)
+                .build()),
+        1);
+  }
+
+  /** Logic for search by Name. */
+  private Pair<List<PractitionerWrapper>, Integer> searchName(
+      MultiValueMap<String, String> parameters) {
+    String name = parameters.getFirst("name");
+    ProviderResponse providerResponse = ppmsClient.providersForName(name);
+    int page = Integer.parseInt(parameters.getOrDefault("page", singletonList("1")).get(0));
+    int count = Integer.parseInt(parameters.getOrDefault("_count", singletonList("15")).get(0));
+    int fromIndex = Math.min((page - 1) * count, providerResponse.value().size());
+    int toIndex = Math.min((fromIndex + count), providerResponse.value().size());
+    /** Retrieve a list of providerResponse from PPMS using the name's identifier. */
+    List<ProviderResponse.Value> providerResponsePages =
+        providerResponse.value().subList(fromIndex, toIndex);
+    /** Using providerResponse, retrieve a list of providerContactsResponse from PPMS. */
+    List<ProviderContactsResponse> providerContactsResponsePages =
+        providerResponsePages
+            .parallelStream()
+            .map(
+                prv -> {
+                  return ppmsClient.providerContactsForId(prv.providerIdentifier().toString());
```

Comment:
This can also be simplified a little.

.map(prv -> ppmsClient.providerContactsForId(prv.providerIdentifier().toString())).collect(Collectors.toList());